.travis.yml

```diff
@@ -21,3 +22,8 @@ script:
     - if [ "$JOB_TYPE" = compile_and_integration_tests_slow ]; then
         ( cd tests && bash run.sh --runslow );
       fi
+
+    - if [ "$JOB_TYPE" = compile_and_integration_tests_amz_s3 ]; then
+        ( cd tests && bash run.sh --runs3 );
+      fi
```

Comment:
If what you're saying is true about the credentials (and if it weren't it would be very insecure), then perhaps we should add some logic here to detect whether we're running a PR build or a master build.